subsys/llext/llext_handlers.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2024 Arduino SA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <zephyr/llext/llext.h>
+#include <zephyr/llext/loader.h>
+#include <zephyr/internal/syscall_handler.h>
+#include <zephyr/kernel.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(llext, CONFIG_LLEXT_LOG_LEVEL);
+
+#include "llext_priv.h"
+
+ssize_t z_impl_llext_get_init_array(struct llext *ext, void *buf, size_t buf_size)
+{
+	if (!ext) {
+		return -EINVAL;
+	}
+
+	size_t preinit_size = ext->mem_size[LLEXT_MEM_PREINIT];
+	size_t init_size = ext->mem_size[LLEXT_MEM_INIT];
+	size_t total_size = preinit_size + init_size;
+
+	if (buf) {
+		char *byte_ptr = buf;
+		const char **fn_ptrs = buf;
```

Comment:
`const void **`? I had to think why `char`, thought there would be some particular purpose in it. Same for `text_start` and `test_end` below. In fact personally I prefer `uint8_t *` over `char *` in such cases for the 2 reasons: (1) these aren't characters or strings, and (2) `uint8_t` has a well defined signedness unlike `char`, but I do realise, that `char` is used a lot in such cases...